src/auth/user.repository.ts

```diff
@@ -86,7 +48,110 @@ export class UserRepository extends Repository<User> {
     return null;
   }
 
-  private async hashPassword(password: string, salt: string): Promise<string> {
+  public async approveCustomer(
+    id: number,
+    user: User,
+  ): Promise<User> {
+    if (user.userType === UserType.admin) {
+      await this.createQueryBuilder()
+        .update(User)
+        .set({ approved: true })
+        .where({ id })
+        .execute();
+
+      const approvedUser: User = await getManager()
+        .createQueryBuilder(User, 'user')
+        .where({ id })
+        .getOne();
+
+      return approvedUser;
+    }
+    throw new UnauthorizedException('Only Admins can approve a user.');
+  }
+
+  public static async getAllTripOrganizers(
+    user: User,
+    params: Params,
+  ): Promise<User[]> {
+    if (user.userType === UserType.admin) {
+      let users: User[];
+
+      if (params.approved === 'true') {
+        users = await getManager()
+          .createQueryBuilder(User, 'user')
+          .where({ userType: UserType.tripOrganizer, approved: true })
+          .getMany();
+
+        return users;
+      }
+
+      if (params.approved === 'false') {
+        users = await getManager()
+          .createQueryBuilder(User, 'user')
+          .where({ userType: UserType.tripOrganizer, approved: false })
+          .getMany();
+
+        return users;
+      }
+      users = await getManager()
+        .createQueryBuilder(User, 'user')
+        .getMany();
+
+      return users;
+    }
+    throw new UnauthorizedException('Only admins can get users information.');
+  }
+
+  public static async signUp(signUpCredentialsDto: SignUpCredentialsDto): Promise<User> {
+    const user: User = new User();
+    user.userType = signUpCredentialsDto.isTripOrganizer ? UserType.tripOrganizer : UserType.customer;
+
+    if (user.userType === UserType.tripOrganizer) {
+      user.facebookId = signUpCredentialsDto.facebookId;
+      user.instagramId = signUpCredentialsDto.instagramId;
+    }
+
+    user.username = signUpCredentialsDto.username;
+    user.email = signUpCredentialsDto.email;
+    user.approved = !signUpCredentialsDto.isTripOrganizer;
+    user.phoneNumber = signUpCredentialsDto.phoneNumber;
+    user.firstName = signUpCredentialsDto.firstName;
+    user.lastName = signUpCredentialsDto.lastName;
+    user.dateOfBirth = signUpCredentialsDto.dateOfBirth;
+    user.gender = signUpCredentialsDto.gender;
+    user.country = signUpCredentialsDto.country;
+    user.city = signUpCredentialsDto.city;
+    user.homeAddress = signUpCredentialsDto.homeAddress;
+    user.salt = await bcrypt.genSalt();
+    user.password = await UserRepository.hashPassword(signUpCredentialsDto.password, user.salt);
+
+    try {
+      await user.save();
+    } catch (error) {
+      if (error.constraint === 'UQ_e12875dfb3b1d92d7d7c5377e22') {
```

Comment:
i think the error constrains are better to be set as consts or enums instead of being hardcoded here

Reply:
done